yoyodyne/datasets.py

```diff
@@ -82,6 +82,7 @@ def __init__(
         self.config = config
         self.samples = list(self.config.samples(filename))
         self.index = index if index is not None else self._make_index()
+        assert self.config.has_features is self.index.has_features
```

Comment:
`==` not `is`?

Reply:
Turns out we have to do neither. `pytest` only sees self.index as a property object so this assertion check fails.